VRFYP2019/Not-Dodgeball
# Add game manager and ball pool

**Description**
First draft of a gameManager, currently just handling restarting the game and game duration. After time is up, press trigger on either hand to restart.
Also add a ball pool that can instantiate new balls dynamically if required. Dead balls are added back to the queue.
We can refine "dead" in the future but right now it's balls that move <0.14f in velocity.

Also the gameManager will need references to all score boards in the scene. Also after this is merged, any scene MUST have the gameManager in it.

@lyhvictoria

**Impact**

**Risks**
Touched several prefabs

**Test Plan**
1. Open the GameMangerScene scene, reduce the game duration (in the inspector for GameManager) and try restarting.
2. Add/remove balls from the queue before starting the game, try spamming spawning, or adjust some parameters like `numOfStartingBalls` in `BallManager.cs`, no errors should show up
